test/library/packages/OrderedSet/general/OsetTest.chpl

```diff
@@ -14,9 +14,9 @@ class testClass {
   var dummy: int = 0;
   proc init(dummy: int=0) { this.dummy = dummy; }
 }
-proc <(lhs: testClass, rhs: testClass) {
+operator testRecord.<(lhs: testClass, rhs: testClass) {
```

Comment:
This and the following has the wrong receiver type.

I wonder how/weather this compiles, though. Here, are we making `<` associated with `testRecord` (so if the type is in scope, so is `<`) but then defining it to accept `testClass` instances? That seems to be like that's what's happening and it is the only way this could still be working, right?

If so, should we prevent this? i.e if you are defining an operator method, at least one of its arguments should be of same type as the receiver?

Reply:
> I wonder how/weather this compiles, though. Here, are we making < associated with testRecord (so if the type is in scope, so is <) but then defining it to accept testClass instances? That seems to be like that's what's happening and it is the only way this could still be working, right?

It does compile, for the reasons you state, yeah.

> If so, should we prevent this? i.e if you are defining an operator method, at least one of its arguments should be of same type as the receiver?

That's a great question.  I don't know that it has to be, since the type on which it is defined is not used within the body of the operator.  However, I'm not sure what value associating it with an uninvolved type provides - it means you could get access to it by importing the associated type, but maybe that counts as hijacking, since it could be unexpected?

Reply:
Yeah, I guess I can see arguments for both allowing and prohibiting. Though I think I am leaning towards preventing it.

- I can't think of a compelling use case for supporting it.
- Kind of similar but, I don't think we could have a similar behavior in the world we were living in before this `operator` effort. (Could we have `proc type` before, or was that just a proposal before the `operator` keyword was accepted as the way to go?) If we couldn't have anything like this before, then preventing it now doesn't set us back in terms of functionality.
- As you mentioned, this could lead to some hijacking, which in turn can cause some errors that are hard to track?

Probably obvious but, I am not advocating for any change regarding this in this PR.

Reply:
I opened #17514 to discuss this